### ISSUE TITLE
rsync-auto sets owner/group in excluded folder

### ISSUE THREAD

**Issue by etienneq** [mitchellh/vagrant#5256]
Hi,

when using synced folders with rsync, I've got the following problem. Owner and group are set in excluded folders too. One has to note that I did not exclude the folder itself but only its content.

I would expect that the chown does not traverse into excluded folders.

Synced folders are configured like this:
```
type: "rsync",
owner: "xyz", group: "xyz",
rsync__auto: true,
rsync__exclude: [ ".git/", "myfolder/*" ],
rsync__args: ["--verbose", "--archive", "--delete", "-z"]
```

Cheers
Etienne

**Comment by ezpuzz** [comment]
This is causing issues on our systems, because the `find` command issued to change the permissions of rsync'd files is giving `No such file or directory` errors for `"tmp/"` files in our project directory.

Command:
```
find '/vagrant' '!' -type l -a '(' ! -user vagrant -or ! -group vagrant ')' -print0 | xargs -0 -r chown vagrant:vagrant
```

Error:
```
find: `/vagrant/tmp/custom_static_compiler-tmp_dest_dir-464MbVTn.tmp': No such file or directory
```

Vagrantfile:
```
config.vm.synced_folder ".","/vagrant", type: "rsync", rsync__exclude: ["node_modules", "bower_components", "dist", "tmp/"]
```

Expected behavior:
The find Command above ignores the directories/files passed to `rsync__exclude` in the Vagrantfile such that the Command becomes:
```
find '/vagrant' '!' -type l -a '(' ! -user vagrant -or ! -group vagrant ')' -not \( -path /vagrant/node_modules -prune \) -not \( -path /vagrant/bower_components -prune \) -not \( -path /vagrant/dist -prune \) -not \( -path /vagrant/tmp -prune \) -print0
```

**Comment by vlopato** [comment]
Hi, All!

Any news with this?

**Comment by niamleeson** [comment]
Has this issue been fixed yet? Running into the same issue

`ERROR -- : exception while processing events: The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

find '/vagrant' '!' -type l -a '(' ! -user vagrant -or ! -group vagrant ')' -print0 | xargs -0 -r chown vagrant:vagrant

Stdout from the command:



Stderr from the command:

find: ‘/vagrant/frontend/tmp/build.lock’: No such file or directory
chown: cannot access ‘/vagrant/frontend/tmp/build.lock’: No such file or directory
 Backtrace:
 -- C:/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/communicators/ssh/communicator.rb:236:in `execute'
 -- C:/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/communicators/ssh/communicator.rb:246:in `sudo'
 -- C:/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/guests/linux/cap/rsync.rb:24:in `rsync_post'
 -- C:/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/capability_host.rb:111:in `call'
 -- C:/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/capability_host.rb:111:in `capability'
 -- C:/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/guest.rb:43:in `capability'
 -- C:/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/synced_folders/rsync/helper.rb:171:in `rsync_single'
 -- C:/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/synced_folders/rsync/command/rsync_auto.rb:189:in `block (2 levels) in callback'
 -- C:/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/synced_folders/rsync/command/rsync_auto.rb:178:in `each'
 -- C:/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/synced_folders/rsync/command/rsync_auto.rb:178:in `block in callback'
 -- C:/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/synced_folders/rsync/command/rsync_auto.rb:177:in `each'
 -- C:/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/synced_folders/rsync/command/rsync_auto.rb:177:in `callback'
 -- C:/vagrant/embedded/gems/gems/listen-3.0.5/lib/listen/event/config.rb:23:in `call'
 -- C:/vagrant/embedded/gems/gems/listen-3.0.5/lib/listen/event/config.rb:23:in `call'
 -- C:/vagrant/embedded/gems/gems/listen-3.0.5/lib/listen/event/processor.rb:115:in `_process_changes'
 -- C:/vagrant/embedded/gems/gems/listen-3.0.5/lib/listen/event/processor.rb:19:in `block in loop_for'
 -- C:/vagrant/embedded/gems/gems/listen-3.0.5/lib/listen/event/processor.rb:15:in `loop'
 -- C:/vagrant/embedded/gems/gems/listen-3.0.5/lib/listen/event/processor.rb:15:in `loop_for'
 -- C:/vagrant/embedded/gems/gems/listen-3.0.5/lib/listen/event/loop.rb:84:in `_wait_for_changes'
 -- C:/vagrant/embedded/gems/gems/listen-3.0.5/lib/listen/event/loop.rb:42:in `block in setup'
 -- C:/vagrant/embedded/gems/gems/listen-3.0.5/lib/listen/internals/thread_pool.rb:6:in `call'
 -- C:/vagrant/embedded/gems/gems/listen-3.0.5/lib/listen/internals/thread_pool.rb:6:in `block in add'
`

**Comment by chrisroberts** [comment]
I have been unable to replicate this behavior. @niamleeson can you share the vagrant file that you're using where you're encountering this error? Have you tried upgrading to 1.8.5, and if so are you still experiencing the same behavior? Thanks!

**Comment by niamleeson** [comment]
The problem was fixed by adding `rsync__chown: false` to Vagrantfile. It was driving me crazy for the longest time.

**Comment by kitsunde** [comment]
The exact same issue happens for me sporadically with EmberJS. Ember also uses a tmp directory (that I have as an excluded sub folder) and sometimes chown will fail because the file no longer exists or seems otherwise to no longer exists.

I'll paste my output and Vagrantfile the next time it happens, but I'm fairy certain it will happen on any Vagranfile setup with rsync, chown and exclude folders.

On Tue, 16 Aug 2016 at 10:30 PM Chris Roberts

<
mailto:Chris Roberts <notifications@github.com>
> wrote:

a, pre, code, a:link, body { word-wrap: break-word !important; }

I have been unable to replicate this behavior.
https://github.com/niamleeson
can you share the vagrant file that you're using where you're encountering this error? Have you tried upgrading to 1.8.5, and if so are you still experiencing the same behavior? Thanks!

—

You are receiving this because you are subscribed to this thread.

Reply to this email directly,
https://github.com/mitchellh/vagrant/issues/5256#issuecomment-240119452
, or

**Comment by niamleeson** [comment]
@kitsunde Try adding `rsync__chown: false` to your Vagrantfile. My problem went away 100%. I am also running Ember.

**Comment by kitsunde** [comment]
Ah I understand what that does now and why it would work, thanks. :)

On Tue, 16 Aug 2016 at 10:45 PM Jay

<
mailto:Jay <notifications@github.com>
> wrote:

a, pre, code, a:link, body { word-wrap: break-word !important; }

https://github.com/kitsunde
Try adding

rsync__chown: false

to your Vagrantfile. My problem went away 100%. I am also running Ember.

—

You are receiving this because you were mentioned.

Reply to this email directly,
https://github.com/mitchellh/vagrant/issues/5256#issuecomment-240124127
, or

**Comment by chrisroberts** [comment]
Are all hosts being used windows?

**Comment by kitsunde** [comment]
I'm on OSX

On Wed, 17 Aug 2016 at 2:47 AM Chris Roberts

<
mailto:Chris Roberts <notifications@github.com>
> wrote:

a, pre, code, a:link, body { word-wrap: break-word !important; }

Are all hosts being used windows?

—

You are receiving this because you were mentioned.

Reply to this email directly,
https://github.com/mitchellh/vagrant/issues/5256#issuecomment-240199530
, or

**Comment by chrisroberts** [comment]
Ah, okay. I see how this is happening. Thanks!

**Comment by chrisroberts** [comment]
Fixed via #7726. Thanks for the report!
